File: Leetcode/Easy/Intersection of Two Arrays II/intersection_of_two_arrays_1.go
// Accepted  4 ms  3.5 MB

func intersect(nums1 []int, nums2 []int) []int {
	var result []int

	map1 := make(map[int]int)
	map2 := make(map[int]int)

	for _, v := range nums1 {
		map1[v]++
	}

	for _, v := range nums2 {
		map2[v]++
	}

	for k, v1 := range map1 {
		if v2, ok := map2[k]; ok {
			for i := 0; i < min(v1, v2); i++ {
				result = append(result, k)
			}
		}
	}

	return result
}

File: Leetcode/Easy/Intersection of Two Arrays/Intersection of Two Arrays_1.cpp
// Accepted  11 ms  11.56 MB

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        int l1 = nums1.size();
        int l2 = nums2.size();
        map<int, bool> map1;
        map<int, bool> map2;
        vector<int> output;

        for (int i = 0; i < l1; i++) {
            map1[nums1[i]] = true;
        }

        for (int i = 0; i < l2; i++) {
            map2[nums2[i]] = true;
        }

        for (map<int, bool>::iterator it = map1.begin(); it != map1.end(); it++) {
            if (map2[it->first]) {
                output.push_back(it->first);
            }
        }

        return output;
    }
};